network-api/networkapi/buyersguide/migrations/0028_uplift_tri_to_quadstate.py

```diff
@@ -0,0 +1,125 @@
+# Generated by Django 2.2.5 on 2019-10-09 21:15
+
+from django.db import migrations
+import networkapi.buyersguide.fields
+
+def tri_to_quad(input):
+    if input == True:
+        return 'Yes'
+    if input == False:
+        return 'No'
+    return 'U'
+
+def copy_tri_to_quad(apps, schema_editor):
+    Product = apps.get_model("buyersguide", "Product")
+    for product in Product.objects.all():
+        product.manage_security_temp = tri_to_quad(product.manage_security)
+        product.must_change_default_password_temp = tri_to_quad(product.must_change_default_password)
+        product.security_updates_temp = tri_to_quad(product.security_updates)
+        product.uses_encryption_temp = tri_to_quad(product.uses_encryption)
+        product.save()
+
+def quad_to_tri(input):
+    if input == 'Yes':
+        return True
+    if input == 'No':
+        return False
+    return None
+
+def copy_quad_to_tri(apps, schema_editor):
+    Product = apps.get_model("buyersguide", "Product")
+    for product in Product.objects.all():
+        product.manage_security = quad_to_tri(product.manage_security_temp)
+        product.must_change_default_password = quad_to_tri(product.must_change_default_password_temp)
+        product.security_updates = quad_to_tri(product.security_updates_temp)
+        product.uses_encryption = quad_to_tri(product.uses_encryption_temp)
+        product.save()
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('buyersguide', '0027_null_boolean_field'),
+    ]
+
+    operations = [
```

Comment:
so this is a four stage uplife:
1. build new fields, with unique names, that use the new 4-state field type
2. migrate the 3-state data from the old fields as 4-state data to the new fields
3. delete the old fields
4. rename the new fields so they use the names of the old fields

This means that at the end of this migrations, nothing has really changed except the fields types got changed from NullBoolean to ExtendedYesNo, and some of the fields got renamed as per the spreadsheet